src/ProgramBlockEditor.js

```diff
@@ -420,7 +506,10 @@ class ProgramBlockEditor extends React.Component<ProgramBlockEditorProps, Progra
         contents.push(this.makeEndOfProgramAddNodeSection(this.props.program.length));
 
         return (
-            <div className='ProgramBlockEditor__container'>
+            <div
+                className='ProgramBlockEditor__container'
+                ref={this.programBlockEditorRef}
```

Comment:
This `ref` needs to be moved to the element on which the `onDragOver`, `onDragLeave`, and `onDrop` events are registered: `ProgramBlockEditor__program-sequence-scroll-container`. Otherwise the source of the `onDragLeave` events will not line up with the bounds calculation in `handleDragLeaveOnProgramArea()`. This results in the highlight remaining in place when it should not. For example, drag a block into the program area, and then leave with it still 'in hand' through the left hand side: the first add node will remain highlighted even though we have left the drop area.

Reply:
Ah, looks like there's already a `ref` on `ProgramBlockEditor__program-sequence-scroll-container`:

    ref={this.programSequenceContainerRef}

I missed it due to the way the code is formatted.

We can use this `ref` in the leave event bounds calculation.

Reply:
Whatever we use as the `ref`, IMO it should be on the next line to avoid further confusion.

Reply:
Yeah, agreed on the `ref` placement.